### ISSUE TITLE
Tensorflow Implementation of Focal Frequency Loss

### ISSUE THREAD

**Issue by ZohebAbai** [EndlessSora/focal-frequency-loss#8]
As I couldn't find a tensorflow implementation of Focal Frequency Loss, so I created it. 

Please visit the [Github Repo](https://github.com/ZohebAbai/tf-focal-frequency-loss) and [PyPi Project](https://pypi.org/project/tf-focal-frequency-loss/).

Use case notebook is included in the Repo. Any feedback is appreciated.

@EndlessSora If you find this implementation useful, kindly do mention it on your README. 

Thanks for releasing this.

**Comment by EndlessSora** [comment]
Thank you so much for the nice TensorFlow implementation! Have you tested the code and its performance compared with this official repo?

**Comment by ZohebAbai** [comment]
I have tested the code for sure. The Tensorflow version gives an exact numerical answer as the Pytorch version. Code for test attached  below:

```python
from focal_frequency_loss import FocalFrequencyLoss as FFL
from tf_focal_frequency_loss import FocalFrequencyLoss as TFFL

# Pytorch
import torch

ffl = FFL(loss_weight=1.0, alpha=1.0)  # initialize nn.Module class

fake = torch.randn(4, 3, 64, 64)
real = torch.randn(4, 3, 64, 64)

pt_loss = ffl(fake, real)  # calculate focal frequency loss


# Tensorflow
import tensorflow as tf

tffl = TFFL(loss_weight=1.0, alpha=1.0)  # initialize tf.keras.layers.Layer class

fake = tf.convert_to_tensor(fake.numpy(), tf.float32)
real = tf.convert_to_tensor(real.numpy(), tf.float32)

tf_loss = tffl(fake, real)  # calculate focal frequency loss

# Check
import numpy as np
assert np.isclose(pt_loss.numpy(), tf_loss.numpy()) , "Results don't match"
```

I have trained a Pix2Pix model and it does give better results (compared to training without it), but I haven't recreated the official repo results to compare its performance.

**Comment by EndlessSora** [comment]
Thanks a lot for the nice support. I have already mentioned this unofficial TensorFlow implementation in our README.

**Comment by ZohebAbai** [comment]
Thanks a lot for the support.
